main backend file.py

```diff
@@ -0,0 +1,39 @@
+
+from bottle import Bottle, run, request, template
+
+
+app = Bottle()
+
+@app.route('/', method = 'POST')
+def post_redirect():
+    global data
+    data = request.json
+    data = data['vals']
+    allints = True
+    for item in data:
+        if not item.isnumeric():
+            allints = False
+    if allints:
+        if len(data) == 500:
+            data.sort()
+            return "success"
+        else:
+            
+            if len(data) < 500:
+                data = ["less than 500 integers - zach"]
+                return "length is less than 500 integers - zach"
+            
+            else:
+                data = ["greater than 500 integers - zach"]
+                return "length is more than 500 integers - zach"
+    else:
+        data = ["not all of the data consists of integers -zach"]
+        return "not all of the data consists of integers -zach"
+@app.route('/', method = 'Get')
+def get_redirect():
+    ret_str = ""
+    for i in data:
+        ret_str += i + " "
+    return ret_str

```

Comment:
Is this String JSON formatted?  I believe the challenge asked for a JSON formatted list.  Would you be able to JSON format the responses?  You can rely on a Python library to manage this for you.

Reply:
You're right. I did not originally have it as a JSON formatted list. I think this way of rewriting the method should take care of it:

@app.route('/', method = 'Get')
def get_redirect():
    json_data = {'vals': data}
    return json_data

run(app, host = 'localhost', port = 8080)

Reply:
Here's a photo of the output
![image](https://user-images.githubusercontent.com/50681932/57953808-c3733880-78b6-11e9-89e5-e61700fd5407.png)